Tests/OpenAITests/OpenAITestsDecoder.swift

```diff
@@ -235,12 +271,16 @@ class OpenAITestsDecoder: XCTestCase {
             created: 1677652288,
             model: .gpt3_5Turbo,
             choices: [
-                .init(index: 0, message:
-                        Chat(role: .assistant,
-                             functionCall: ChatFunctionCall(name: "get_current_weather", arguments: nil)),
-                      finishReason: "function_call")
+                .init(
+                    index: 0,
+                    // TODO: Fix "Ambiguous use of 'init(role:content:name:functionCall:)'" when omitting content
+                    // maybe by removing `init` the overload?
+                    message: Chat(role: .assistant, content: nil as String?, functionCall: .init(name: "get_current_weather", arguments: nil)),
```

Comment:
I tried to keep existing constructors for backwards compatibility but this ended up like this
`content: nil as String?` . If you try to omit the content it returns error because is too generic

Reply:
I got the issue, it was in ChatQuery having a lot of inits. What do you think about reducing it to:
```swift
   public init(role: Role, content: Codable? = nil, name: String? = nil, functionCall: ChatFunctionCall? = nil) {
        let stringOrCodable: StringOrCodable<[ChatContent]>?
        
        if let string = content as? String {
            stringOrCodable = .string(string)
        } else if let arr = content as? [ChatContent] {
            stringOrCodable = .object(arr)
        } else {
            stringOrCodable = nil
        }
        
        self.init(role: role, contents: stringOrCodable, name: name, functionCall: functionCall)
   }
```
And with this kind of constructor, the error you mentioned here will be solved and you'll be able to omit `content` like this:
```swift
Chat(role: .assistant, functionCall: .init(name: "get_current_weather", arguments: nil)
```

Reply:
awesome i'll try this asap

Reply:
fixed in  https://github.com/MacPaw/OpenAI/pull/122/commits/791484c93590f48e11d65f42c80a3258bc3ea9ea